cartographer_ros/cartographer_ros/node.cc

```diff
@@ -725,6 +729,39 @@ bool Node::HandleReadMetrics(
   return true;
 }
 
+bool Node::HandleDeleteTrajectory(
+    cartographer_ros_msgs::DeleteTrajectory::Request& request,
+    cartographer_ros_msgs::DeleteTrajectory::Response& response) {
+  absl::MutexLock lock(&mutex_);
+  // We can only delete FINISHED or FROZEN trajectories.
+  response.status = TrajectoryStateToStatus(
+      request.trajectory_id,
+      {TrajectoryState::FINISHED, TrajectoryState::FROZEN} /* valid_states */);
+  if (response.status.code != cartographer_ros_msgs::StatusCode::OK) {
+    LOG(ERROR) << "Can't delete trajectory: " << response.status.message;
+    return true;
+  }
+  map_builder_bridge_.DeleteTrajectory(request.trajectory_id);
+  response.status.message = absl::StrCat(
+      "Scheduled trajectory ", request.trajectory_id, " for deletion.");
+  LOG(INFO) << response.status.message;
+  return true;
+}
+
+bool Node::HandleLoadStateFromFile(
+    cartographer_ros_msgs::LoadStateFromFile::Request& request,
+    cartographer_ros_msgs::LoadStateFromFile::Response& response) {
+  absl::MutexLock lock(&mutex_);
+  // TODO(MichaelGrupp): this crashes if the file is invalid.
```

Comment:
why not handle the case and return it with the status?

Should be easy with `try / catch` rigth?